src/net.cpp

```diff
@@ -2864,13 +2865,16 @@ bool CConnman::ForNode(NodeId id, std::function<bool(CNode* pnode)> func)
     return found != nullptr && NodeFullyConnected(found) && func(found);
 }
 
-int64_t CConnman::PoissonNextSendTo(int64_t now, int average_interval_seconds, uint64_t net_group)
+int64_t CConnman::PoissonNextSendTo(int64_t now, bool inbound, uint64_t net_group)
 {
-    std::pair<int, uint64_t> key{average_interval_seconds, net_group};
+    if (!inbound) {
+      return PoissonNextSend(now, INVENTORY_BROADCAST_INTERVAL >> 1);
```

Comment:
I think it's suboptimal to put knowledge of the actual scheduling policy in net; it belongs more in net_processing (the `INVENTORY_BROADCAST_INTERVAL` and `NETWORK_GROUP_BUCKETS` should move there as well, I think).

How about passing a boolean `per_group` in, or saying that when `net_group = 0` there is no grouping to be done? Then you can keep the decision logic in net_processing, and keep this function generally usable.